### PR TITLE
[ENH] to_orc

### PR DESCRIPTION
**Pull request by ghost**
Add pandas.io.orc.to_orc method definition

### CODE REVIEW

**Comment by debnathshoham** [comment]
Thanks for the PR @NickFillot!
Is there an existing pandas issue number you're trying to address?
Do you mind creating one, if there isn't one?

**Comment by NickFillot** [comment]
> Thanks for the PR @NickFillot! Is there an existing pandas issue number you're trying to address? Do you mind creating one, if there isn't one?

Just created one @ [to_orc Issue](https://github.com/pandas-dev/pandas/issues/43864) didn't see one related to it

Thank you

**Comment by jreback** [comment]
tests pls

follow the existing way we test to_parquet for example with the fixtures that skip based in thr version

**Review by jreback — CHANGES_REQUESTED** [review]
see comments

**Comment by alimcmaster1** [comment]
Thanks for the PR @NickFillot comments above!

**Comment by NickFillot** [comment]
Working on tests, i'm trying to understand how pandas testing works

**Comment by iajoiner** [comment]
@NickFillot Thanks for working on this! Note that your ordering actually doesn't work for **write_table** in pyarrow 4.0.0 so please either use the path, table ordering to accommodate that version or set the minimum version of pyarrow to 4.0.1.

**Review by jreback — CHANGES_REQUESTED** [review]
pls add tests.

**Comment by iajoiner** [comment]
@NickFillot Do you mind me reopening it?

File: pandas/io/orc.py [hunk 1]
```diff
@@ -55,3 +59,68 @@ def read_orc(
...
+    else:
+        try:
+            assert engine.__name__ == 'pyarrow', "engine must be 'pyarrow' module"
+            assert hasattr(engine, 'orc'), "'pyarrow' module must have orc module"
+        except Exception as e:
+            raise ValueError("Wrong engine passed, %s" % e)
+            
+    if path is None:
+        # to bytes: tmp path, pyarrow auto closes buffers
+        with tm.ensure_clean(os.path.join(gettempdir(), os.urandom(12).hex())) as path:
```
[the remark this answers is not in the document]
Comment (iajoiner) [reply pandas/io/orc.py:114]: Yes I do close the file-like object from my side by default in Arrow. It does seem to be different from the behavior of the Parquet writer in Arrow. If this is indeed an issue I can discuss with the Arrow community whether we should change it.

Right now I use PyArrow buffer and avoid creating a temp file.

File: pandas/io/orc.py [hunk 2]
```diff
@@ -55,3 +59,68 @@ def read_orc(
...
+    if index is None:
+        index = df.index.names[0] is not None
+    
+    if isinstance(engine, str):
+        engine = import_optional_dependency(engine, min_version='4.0.0')
+    else:
+        try:
+            assert engine.__name__ == 'pyarrow', "engine must be 'pyarrow' module"
+            assert hasattr(engine, 'orc'), "'pyarrow' module must have orc module"
+        except Exception as e:
```
[the remark this answers is not in the document]
Comment (iajoiner) [reply pandas/io/orc.py:109]: Fixed

**Comment by iajoiner** [comment]
This PR has been reopened as https://github.com/pandas-dev/pandas/pull/44554

**Pull request closed without merging**